src/metal.cs

```diff
@@ -699,6 +793,94 @@ partial interface MTLDevice {
 #endif
 		[Export ("supportsTextureSampleCount:")]
 		bool SupportsTextureSampleCount (nuint sampleCount);
+
+		[Mac (10, 13), NoiOS, NoWatch, NoTV]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("removable")]
+		bool Removable { [Bind ("isRemovable")] get; }
+
+		[Mac (10, 13), NoiOS, NoWatch, NoTV]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("readWriteTextureSupport")]
+		MTLReadWriteTextureTier ReadWriteTextureSupport { get; }
+
+		// @required @property (readonly) MTLArgumentBuffersTier argumentBuffersSupport __attribute__((availability(macos, introduced=10_13)));
+		[Mac (10, 13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("argumentBuffersSupport")]
+		MTLArgumentBuffersTier ArgumentBuffersSupport { get; }
+
+		// @required @property (readonly, getter = areRasterOrderGroupsSupported) BOOL rasterOrderGroupsSupported __attribute__((availability(macos, introduced=10_13)));
+		[Mac (10, 13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("rasterOrderGroupsSupported")]
+		bool RasterOrderGroupsSupported { [Bind ("areRasterOrderGroupsSupported")] get; }
+
+		[Mac (10,13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("newLibraryWithURL:error:")]
+		[return: NullAllowed]
+		IMTLLibrary CreateLibrary (NSUrl url, [NullAllowed] out NSError error);
+
+		[Mac (10,13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("minimumLinearTextureAlignmentForPixelFormat:")]
+		nuint GetMinimumLinearTextureAlignment (MTLPixelFormat format);
+
+		[Mac (10,13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("maxThreadgroupMemoryLength")]
+		nuint MaxThreadgroupMemoryLength { get; }
+
+[Mac (10,13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("programmableSamplePositionsSupported")]
+		bool ProgrammableSamplePositionsSupported { [Bind ("areProgrammableSamplePositionsSupported")] get; }
+
+		[Mac (10,13), iOS (11,0), TV (11,0), NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("getDefaultSamplePositions:count:")]
+		void GetDefaultSamplePositions (IntPtr positions, nuint count);
```

Comment:
to resume: this is an array of a struct inside a protocol, right ?
@rolfbjarne can the registrar handle this ?